### PR TITLE
Add bootstrap popover to authors shown in header

### LINKED ISSUE #6923: Add popover to list authors on wiki page headings
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### What is to be done.
[Wiki pages](https://publiclab.org/wiki/onecranston) now have this great header showing Authors, Revisions, and the talk page:

image

Let's make it so you can click the Authors item and see a [Bootstrap popover](https://getbootstrap.com/docs/4.3/components/popovers/) listing the authors' usernames.

The code for that is here:

https://github.com/publiclab/plots2/blob/6d417a384801e4c20b77c4d6360788f19a78b41c/app/views/wiki/_header.html.erb#L34-L36

We can get the authors listing using `node.authors`, as shown here:

https://github.com/publiclab/plots2/blob/6d417a384801e4c20b77c4d6360788f19a78b41c/app/models/node.rb#L278-L280

And for each one, we could do the following so that their names are clickable and lead to their profile pages!

```html
<ul>
  <% node.authors.each do |author| %>
  <li><a href="/profile/<%= author.username %>">@<%= author.username %></a></li>
  <% end %>
</ul>
```
That code can go inside the popover!

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [$FILENAME]($BRANCH_URL) in the `$REPO` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
$DIFF
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by takitsuse**
Trying to fix #6923

Here are the current screencasts showing the behavior of author popover.
[popover when with 1 author](https://imgur.com/a/jtcJito)
[popover when with 3 authors](https://imgur.com/a/V97cKoP)

As you see in these screencasts, the popover vanishes when we move the cursor to the authors' links. This is not what we intended but I have no idea how to fix this behavior.

@jywarren @gauravano @SidharthBansal @IshaGupta18 @ananyaarun @pydevsg @sashadev-sky @cesswairimu 
Could you help me out?
Thanks!

### CODE REVIEW

**Comment by cesswairimu** [comment]
Awesome :rocket: thanks @takitsuse,...not sure why travis is failing... Restarting it.. Are the links clickable? Thanks for working on this

**Comment by takitsuse** [comment]
In developer console of Firefox I confirm that they are working correctly. Thank you @cesswairimu

**Comment by sashadev-sky** [comment]
@takitsuse in regard to "As you see in these screencasts, the popover vanishes when we move the cursor to the authors' links. This is not what we intended but I have no idea how to fix this behavior." - that's actually a valid point and thank u for pointing that out!

I pulled your PR and looks like this with multiple authors like @cesswairimu requested
image

I'm thinking what would look better is if you add these attributes to the `<a>` tag:
1. `data-toggle="popover"` 
2. change `data-title` to `data-content`.
3. `tabindex="0"`
4. `data-trigger="focus"`

That would result in:
image

let me know your thoughts :)

**Comment by cesswairimu** [comment]
Love it, thanks @sashadev-sky

**Comment by takitsuse** [comment]
That looks gorgeous! I will make new commit according to your nice suggestion.

**Comment by jywarren** [comment]
Awesome, all ready to merge now? Thanks!!!

File: app/views/wiki/_header.html.erb [hunk 1]
```diff
@@ -31,7 +31,15 @@
...
       <span class="d-none d-lg-inline">
-        <%= number_with_delimiter(@node.authors.length)  %> Authors |
+        <span rel="tooltip" tabindex="0" data-placement="top" data-toggle="popover" data-trigger="focus" data-html="true" data-content="
+          <ul>
+            <% @node.authors.each do |author| %>
+              <li><a href='/profile/<%= author.username %>'>@<%= author.username %></a></li>
+            <% end %>
+          </ul>
+        ">
+          <%= number_with_delimiter(@node.authors.length) %> Authors
```
Comment (seabl) [inline app/views/wiki/_header.html.erb]: Should we _pluralize_ 'Authors' here?

Comment (SidharthBansal) [reply app/views/wiki/_header.html.erb]: I hope so

**Comment by jywarren** [comment]
Ah, it looks like this needs some conflicts resolved; some other work happened touching these lines. Can you give it a try? Happy to help!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**